### PR TITLE
Dynamic like v3 kay diversion

### PR DESCRIPTION
**Pull request by henryhobhouse**
Altered JS to allow post submit on Like to allow server side update

### CODE REVIEW

File: app/assets/stylesheets/likes.scss [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
+  text-align: center;
+  color: #fff;
+  background-color: #445561;
+  transition: all .1s;
+}
+
+.like-button:hover {
+  border: solid 2px #fff;
+  // background-color: rgba(255, 255, 255, .4);
+  // box-shadow: 0 0 10px #fff;
```
Comment (neoeno) [inline app/assets/stylesheets/likes.scss]: Commented out code

File: app/assets/javascripts/likes.js [hunk 2]
```diff
@@ -1,2 +1,17 @@
-// Place all the behaviors and hooks related to the matching controller here.
-// All this logic will automatically be available in application.js.
+function ready(fn) {
+  if (document.readyState != 'loading'){
+    fn();
+  } else {
+    document.addEventListener('DOMContentLoaded', fn);
+  }
+}
```
Comment (neoeno) [inline app/assets/javascripts/likes.js]: I'd extract the `ready` function to another file and use rails require comments to require it first in `application.js`. That way we won't have it repeated in `avatar.js` and `likes.js`.

Bear in mind you'll need to test the avatar fnality too to make sure it works!

**Review by neoeno — CHANGES_REQUESTED** [review]
Some great work here! I'm really tempted to just say go ahead but I know I'd be failing you if I didn't give you some feedback to go away with!

**Review by neoeno — APPROVED** [review]
Love this change! Great to see the refactor getting combined with refactoring another feature — nice work!

**Comment by BertZZ** [comment]
@henryhobhouse want to resolve these conflicts?

**Comment by henryhobhouse** [comment]
Update again to finally work with new method to split out the different onload functions in their relative locations so defunct JS isn't constantly being called. Updated so the quotes are working; presumably broken from another update from Tim and Issy's work?

**Pull request merged**